github/github.go

```diff
@@ -986,6 +996,59 @@ func (g *GitHub) UpdateReleasePage(
 	return release, nil
 }
 
+// toRepositoryRelease builds a repository release from the set of options
+func (urpo *UpdateReleasePageOptions) toRepositoryRelease() *github.RepositoryRelease {
+	request := &github.RepositoryRelease{}
+
+	if urpo.Name != "" {
+		request.Name = &urpo.Name
+	}
+	if urpo.Body != "" {
+		request.Body = &urpo.Body
+	}
+
+	request.Draft = &urpo.Draft
+	request.Prerelease = &urpo.Prerelease
+
+	if urpo.Latest {
+		request.MakeLatest = ptr.To("true")
+	} else {
+		request.MakeLatest = ptr.To("false")
+	}
+
+	return request
+}
+
+// UpdateReleasePageWithOptions updates release pages (same as UpdateReleasePage),
+// but does so by taking a UpdateReleasePageOptions parameter. It will _not_ set
+// a release as latest unless the corresponding option is set.
+func (g *GitHub) UpdateReleasePageWithOptions(owner, repo string,
+	releaseID int64,
+	tag, commitish string,
+	opts *UpdateReleasePageOptions,
+) (release *github.RepositoryRelease, err error) {
+	logrus.Infof("Updating release page for %s", tag)
+
+	if opts == nil {
+		opts = &UpdateReleasePageOptions{}
+	}
```

Comment:
I'm wondering if we should fail if there are no options provided. 🤔

Reply:
I was designing this for options to be completely optional, so I would rather address your points above (that things should be bool pointer) instead of failing here.

Reply:
What worries me is that calling `UpdateReleasePageWithOptions` without any option might lead to the undesired behavior. To put it in a different way, why would someone try to update the release without providing any option? For example, if you want to clear the release name and the release description, you can set those two parameters to an empty string in the `opts` struct.

If there's some important use case for allowing `opts` to be `nil`, I strongly recommend documenting it, otherwise, I think returning an error is a safer approach.

Reply:
The confusing bit about `UpdateReleasePage[...]` is that it's also used to _create_ releases as far as I understood. So if you just want to create a release, you also use this method, and as far as I understand it's totally valid to pass none of the fields in the options and simply get a release with `UpdateReleasePageWithOptions`.

If you don't pass any options - and this is something I guess I have to validate - my expectation would be that the release doesn't change at all. But ... I'm not sure anymore now.

Reply:
See https://github.com/kubernetes-sigs/release-sdk/pull/288#discussion_r1430463730 re: pointers.